### PR TITLE
no user with kyc exception

### PR DESCRIPTION
**Pull request by josipkoncurat**

### CODE REVIEW

File: src/main/kotlin/com/ampnet/reportservice/service/impl/TemplateDataServiceImpl.kt [hunk 1]
```diff
@@ -189,4 +191,9 @@ class TemplateDataServiceImpl(
         userUuid: String
     ): List<Transaction> =
         userTransactions[userUuid]?.mapNotNull { TransactionFactory.createTransaction(it) }.orEmpty()
+
+    private fun throwExceptionIfNoUserWithInfo(users: UsersExtendedResponse) {
+        if (users.usersList.isEmpty())
+            throw ResourceNotFoundException(ErrorCode.USER_MISSING_INFO, "No user has went through kyc on the platform")
```
Comment (githinho) [inline src/main/kotlin/com/ampnet/reportservice/service/impl/TemplateDataServiceImpl.kt:197]: This exception will return BadRequest - 400, but we want Not Found - 404.
Message: "None of the users went through KYC and completed at least one transaction."

File: src/main/kotlin/com/ampnet/reportservice/exception/ErrorCode.kt [hunk 2]
```diff
@@ -3,7 +3,7 @@ package com.ampnet.reportservice.exception
 enum class ErrorCode(val categoryCode: String, var specificCode: String, var message: String) {
 
     // Users: 03
-    USER_MISSING_PRIVILEGE("03", "05", "Missing privilege to access data"),
+    USER_MISSING_INFO("03", "07", "No user with info on the platform"),
```
Comment (githinho) [inline src/main/kotlin/com/ampnet/reportservice/exception/ErrorCode.kt:6]: Add error code to docs repo.

**Review by githinho — CHANGES_REQUESTED** [review]

**Review by githinho — CHANGES_REQUESTED** [review]
Add test in `ReportControllerTest` for this case.

**Review by mightymatth — COMMENTED** [review]
Seems fine to me!

**Review by githinho — APPROVED** [review]

**Pull request merged**